commit 1f345b45a5eeaedfcf5c48ac328c32d32d30ac26
Author: Fabio M. Di Nitto <fdinitto@redhat.com>
Date:   Mon Mar 28 16:12:05 2011 +0200

    cman preconfig: fix segfault if we cannot determine cluster name
    
    This takes a lot of effort to reproduce, but it can still happen.
    
    Resolves: rhbz#728230
    
    Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

diff --git a/cman/daemon/cman-preconfig.c b/cman/daemon/cman-preconfig.c
index cde130b..4eb6250 100644
--- a/cman/daemon/cman-preconfig.c
+++ b/cman/daemon/cman-preconfig.c
@@ -1291,6 +1291,11 @@ static int get_cman_globals(struct objdb_iface_ver0 *objdb)
 
 	objdb_get_string(objdb, cluster_parent_handle, "name", &cluster_name);
 
+	if (!cluster_name) {
+		sprintf(error_reason, "Unable to determine cluster name.\n");
+		return -1;
+	}
+
 	/* Get the <cman> bits that override <totem> bits */
 	objdb->object_find_create(cluster_parent_handle, "cman", strlen("cman"), &find_handle);
 	if (objdb->object_find_next(find_handle, &object_handle) == 0) {
